Library imports

In [1]:
import pandas as pd
import numpy as np

The first file is over a period from 2010 until 2016, the second file is from 2001 to 2016. The data comes from NOAA.

In [2]:
weather_chicago = pd.read_csv('C:\Users\Ondrej Bohdal\Downloads\\688303.csv')
weather_chicago2 = pd.read_csv('C:\Users\Ondrej Bohdal\Downloads\\688348.csv')

C:\Users\Ondrej Bohdal\Anaconda2\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (2,3,4) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


We check whether there are some reasonable values in the temperatures.

In [14]:
len(weather_chicago['TMIN'].unique())

150

Here we find the stations which have recorded some values for min temperature, precipitation, wind speed. This is true only for a little of all the stations.

In [4]:
weather_chicago[(weather_chicago['TMIN']!=-9999) & (weather_chicago['PRCP']!=-9999) & (weather_chicago['AWND']!=-9999)]['STATION_NAME'].unique()


array(['CHICAGO PALWAUKEE AIRPORT IL US',
       'CHICAGO OHARE INTERNATIONAL AIRPORT IL US',
       'CHICAGO MIDWAY AIRPORT IL US'], dtype=object)

CHICAGO MIDWAY AIRPORT IL US is the closest airport of the three to the city centre of Chicago. Hence we may take the data from there as corresponding to the values for the weather in Chicago.

In [5]:
weather_chicago2[weather_chicago2['STATION_NAME']=="CHICAGO MIDWAY AIRPORT IL US"]

STATION                  STATION_NAME ELEVATION  LATITUDE  \
301330  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301331  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301332  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301333  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301334  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301335  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301336  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301337  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301338  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301339  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301340  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301341  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301342  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301343  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301344  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301345  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301346  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301347  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301348  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301349  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301350  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301351  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301352  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301353  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301354  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301355  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301356  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301357  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301358  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
301359  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
...                   ...                           ...       ...       ...   
306816  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306817  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306818  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306819  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306820  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306821  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306822  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306823  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306824  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306825  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306826  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306827  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306828  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306829  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306830  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306831  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306832  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306833  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  41.78611   
306834  GHCND:USW00014819  CHICAGO MIDWAY AIRPORT IL US     186.5  4

In [6]:
weather_chicago2[(weather_chicago2['TMIN']!=-9999) & (weather_chicago2['PRCP']!=-9999) & (weather_chicago2['AWND']!=-9999)]

STATION                     STATION_NAME ELEVATION  \
113325  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113326  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113327  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113328  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113329  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113330  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113331  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113332  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113333  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113334  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113335  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113336  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113337  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113338  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113339  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113340  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113341  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113342  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113343  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113344  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113345  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113346  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113347  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113348  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113349  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113350  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113351  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113352  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113353  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
113356  GHCND:USW00004838  CHICAGO PALWAUKEE AIRPORT IL US     193.9   
...                   ...                              ...       ...   
306816  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306817  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306818  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306819  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306820  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306821  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306822  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306823  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306824  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306825  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306826  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306827  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306828  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306829  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306830  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306831  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306832  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306833  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306834  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306835  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306836  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306837  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306838  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US     186.5   
306839  GHCND:USW00014819     CHICAGO MIDWAY AIRPORT IL US

GPS coordinates of Chicago Midway Airport: Latitude, Longitude: 41.78611,-87.75222

In [7]:
cleaner_weather_chicago = weather_chicago[weather_chicago['STATION_NAME']=="CHICAGO MIDWAY AIRPORT IL US"]
cleaner_weather_chicago2 = weather_chicago2[weather_chicago2['STATION_NAME']=="CHICAGO MIDWAY AIRPORT IL US"]

Now we keep only the important parameters - max and min temperature, precipitation, name, wind speed, date.

In [8]:
del cleaner_weather_chicago['TOBS']
del cleaner_weather_chicago['STATION']
del cleaner_weather_chicago['ELEVATION']
del cleaner_weather_chicago['LATITUDE']
del cleaner_weather_chicago['LONGITUDE']
del cleaner_weather_chicago2['TOBS']
del cleaner_weather_chicago2['STATION']
del cleaner_weather_chicago2['ELEVATION']
del cleaner_weather_chicago2['LATITUDE']
del cleaner_weather_chicago2['LONGITUDE']
del cleaner_weather_chicago2['TSUN']

We save the results from the data frame into a csv file.

In [9]:
#cleaner_weather_chicago.to_csv("WeatherChicago20102016.csv")
cleaner_weather_chicago2.to_csv("WeatherChicago20012016.csv")

In [10]:
cleaner_weather_chicago2

STATION_NAME      DATE  PRCP  TMAX  TMIN  AWND
301330  CHICAGO MIDWAY AIRPORT IL US  20010101     0   -33  -106    33
301331  CHICAGO MIDWAY AIRPORT IL US  20010102     0   -67  -139    39
301332  CHICAGO MIDWAY AIRPORT IL US  20010103     0   -22  -139    54
301333  CHICAGO MIDWAY AIRPORT IL US  20010104     0     0   -50    54
301334  CHICAGO MIDWAY AIRPORT IL US  20010105     0    28   -56    59
301335  CHICAGO MIDWAY AIRPORT IL US  20010106     0    17   -67    42
301336  CHICAGO MIDWAY AIRPORT IL US  20010107     0    17   -50    44
301337  CHICAGO MIDWAY AIRPORT IL US  20010108     0   -17   -83    43
301338  CHICAGO MIDWAY AIRPORT IL US  20010109     0   -39  -111    52
301339  CHICAGO MIDWAY AIRPORT IL US  20010110     0    17   -67    47
301340  CHICAGO MIDWAY AIRPORT IL US  20010111     0    44   -44    28
301341  CHICAGO MIDWAY AIRPORT IL US  20010112     0    28   -22    24
301342  CHICAGO MIDWAY AIRPORT IL US  20010113     0    33    11    34
301343  CHICAGO MIDWAY AIRPORT IL US  20010114    58    33     6    44
301344  CHICAGO MIDWAY AIRPORT IL US  20010115     3    22     0    62
301345  CHICAGO MIDWAY AIRPORT IL US  20010116     0     0   -28    44
301346  CHICAGO MIDWAY AIRPORT IL US  20010117     0   -28   -67    37
301347  CHICAGO MIDWAY AIRPORT IL US  20010118     0     0   -72    42
301348  CHICAGO MIDWAY AIRPORT IL US  20010119     0   -17   -72    55
301349  CHICAGO MIDWAY AIRPORT IL US  20010120     0   -33  -100    42
301350  CHICAGO MIDWAY AIRPORT IL US  20010121     0   -22  -111    27
301351  CHICAGO MIDWAY AIRPORT IL US  20010122     0   -11  -106    30
301352  CHICAGO MIDWAY AIRPORT IL US  20010123     0    22   -50    36
301353  CHICAGO MIDWAY AIRPORT IL US  20010124     0   -17   -72    54
301354  CHICAGO MIDWAY AIRPORT IL US  20010125     0   -33   -94    41
301355  CHICAGO MIDWAY AIRPORT IL US  20010126     0     0   -50    59
301356  CHICAGO MIDWAY AIRPORT IL US  20010127     0   -28   -83    56
301357  CHICAGO MIDWAY AIRPORT IL US  20010128     0   -11   -94    33
301358  CHICAGO MIDWAY AIRPORT IL US  20010129   170    61   -11    51
301359  CHICAGO MIDWAY AIRPORT IL US  20010130    41    44     6    41
...                              ...       ...   ...   ...   ...   ...
306816  CHICAGO MIDWAY AIRPORT IL US  20160114     0    72   -21    49
306817  CHICAGO MIDWAY AIRPORT IL US  20160115     5    61     0    57
306818  CHICAGO MIDWAY AIRPORT IL US  20160116     0     0   -99    57
306819  CHICAGO MIDWAY AIRPORT IL US  20160117     0   -99  -171    71
306820  CHICAGO MIDWAY AIRPORT IL US  20160118     0  -121  -188    55
306821  CHICAGO MIDWAY AIRPORT IL US  20160119     0   -77  -166    26
306822  CHICAGO MIDWAY AIRPORT IL US  20160120     8   -38  -105    12
306823  CHICAGO MIDWAY AIRPORT IL US  20160121     0     0   -88    12
306824  CHICAGO MIDWAY AIRPORT IL US  20160122     3     0   -60    51
306825  CHICAGO MIDWAY AIRPORT IL US  20160123     0    17   -43    27
306826  CHICAGO MIDWAY AIRPORT IL US  20160124     0    28   -43    37
306827  CHICAGO MIDWAY AIRPORT IL US  20160125    15    61   -16    47
306828  CHICAGO MIDWAY AIRPORT IL US  20160126     0    33   -27    69
306829  CHICAGO MIDWAY AIRPORT IL US  20160127     0    17   -32    57
306830  CHICAGO MIDWAY AIRPORT IL US  20160128     0    50    -5    65
306831  CHICAGO MIDWAY AIRPORT IL US  20160129     0    28   -43    48
306832  CHICAGO MIDWAY AIRPORT IL US  20160130     0    94    22    38
306833  CHICAGO MIDWAY AIRPORT IL US  20160131     5    94    11    42
306834  CHICAGO MIDWAY AIRPORT IL US  20160201     0    83    -5    20
306835  CHICAGO MIDWAY AIRPORT IL US  20160202   188    44    22    49
306836  CHICAGO MIDWAY AIRPORT IL US  20160203     0    39   -16    60
306837  CHICAGO MIDWAY AIRPORT IL US  20160204     0     0   -60    45
306838  CHICAGO MIDWAY AIRPORT IL US  20160205     0    11   -27    31
306839  CHICAGO MIDWAY AIRPORT IL US  20160206     0    56   -32    37
306840  CHICAGO MIDWAY AIRPORT IL US  20160207  

In [11]:
#weather_chicago[weather_chicago['STATION_NAME']=="CHICAGO OHARE INTERNATIONAL AIRPORT IL US"]

In [12]:
#weather_chicago[weather_chicago['STATION_NAME']=="CHICAGO PALWAUKEE AIRPORT IL US"]

In [13]:
#weather_chicago['STATION_NAME'].unique()